aiohttp/client.py

```diff
@@ -911,9 +975,62 @@ def ws_connect(
                 proxy_headers=proxy_headers,
                 compress=compress,
                 max_msg_size=max_msg_size,
+                decode_text=decode_text,
             )
         )
 
+    @overload
+    async def _ws_connect(
+        self,
+        url: StrOrURL,
+        *,
+        method: str = ...,
+        protocols: Collection[str] = ...,
+        timeout: ClientWSTimeout | _SENTINEL = ...,
+        receive_timeout: float | None = ...,
+        autoclose: bool = ...,
+        autoping: bool = ...,
+        heartbeat: float | None = ...,
+        auth: BasicAuth | None = ...,
+        origin: str | None = ...,
+        params: Query = ...,
+        headers: LooseHeaders | None = ...,
+        proxy: StrOrURL | None = ...,
+        proxy_auth: BasicAuth | None = ...,
+        ssl: SSLContext | bool | Fingerprint = ...,
+        server_hostname: str | None = ...,
+        proxy_headers: LooseHeaders | None = ...,
+        compress: int = ...,
+        max_msg_size: int = ...,
```

Comment:
If we define these centrally with Unpack, then the change may not be so bad..

Reply:
Don't we need Python 3.11 for Unpack?